platform/osgi/platform-osgi-configadmin/src/test/java/org/codice/felix/cm/file/EncryptingWrappedPersistenceManagerTest.java

```diff
@@ -0,0 +1,135 @@
+ */
+package org.codice.felix.cm.file;
+
+import static java.lang.String.format;
+import static org.codice.felix.cm.file.EncryptingWrappedPersistenceManager.FELIX_FILENAME;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.osgi.framework.Constants.SERVICE_PID;
+import static org.osgi.service.cm.ConfigurationAdmin.SERVICE_FACTORYPID;
+
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.Hashtable;
+
+import org.apache.felix.cm.PersistenceManager;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EncryptingWrappedPersistenceManagerTest {
+    private static final String PID = "my_pid";
+
+    private static final String VALUE_TO_ENCRYPT = "some_string";
+
+    private static final Dictionary<String, Object> TEST_CASES = new Hashtable<>();
+
+    static {
+        // We don't encrypt integers
+        TEST_CASES.put("1", 3);
+        // We SHOULD encrypt strings
+        TEST_CASES.put("2", VALUE_TO_ENCRYPT);
+        // Except the following three strings which are special for OSGi / Felix
+        TEST_CASES.put(SERVICE_PID, PID);
+        TEST_CASES.put(SERVICE_FACTORYPID, PID);
+        TEST_CASES.put(FELIX_FILENAME, PID);
+    }
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Mock
+    private PersistenceManager mockManager;
+
+    private EncryptingWrappedPersistenceManager persistenceManager;
+
+    @Before
+    public void setup() throws Exception {
+        String pwDirectory = temporaryFolder.newFolder()
+                .getAbsolutePath();
+        persistenceManager = new EncryptingWrappedPersistenceManager(mockManager, pwDirectory);
+    }
+
+    @Test
+    public void testRoundTrip() throws Exception {
+        ArgumentCaptor<Dictionary> captor = ArgumentCaptor.forClass(Dictionary.class);
+        persistenceManager.store(PID, TEST_CASES);
+        verify(mockManager).store(eq(PID), captor.capture());
+
+        Dictionary<String, Object> capturedProps = captor.getValue();
+        String valueToEncrypt = (String) capturedProps.get("2");
+        assertThat(capturedProps.size(), is(5));
+        assertSpecialStrings(capturedProps);
+
+        assertThat("VALUE_TO_ENCRYPT was not actually encrypted",
+                valueToEncrypt,
+                is(not(VALUE_TO_ENCRYPT)));
+        assertThat("Encrypting VALUE_TO_ENCRYPT results in a super-sequence of the original",
```

Comment:
✏️  Maybe it's just me but the use of the term "super-sequence" is a little fancy for this context. Could just say "...results in a string containing the original".
✏️  Past tense: "...resulted in..."